### PR TITLE
JAX version of Single Plane, Single Plane LOS and SIS

### PR DESCRIPTION
**Pull request by nataliehogg**
This PR includes a basic jaxxed version of the SinglePlane and SinglePlaneLOS classes, as well as updates to LensModel to import the jaxtronomy versions. A list of jaxxed profiles is added to ProfileListBase, with non-jaxxed profiles being imported from lenstronomy. A warning to the user is given when these profiles are called. 

Also added is the SIS profile.

To do: update test_SinglePlaneLOS.

### CODE REVIEW

**Comment by sibirrer** [comment]
@nataliehogg thank you very much! Are you working on this PR? It's quite a big one and we (@ahuang314 and myself) are wondering where Alan can develop in parallel.

**Comment by sibirrer** [comment]
@nataliehogg I think the decoupled planes are in pip, it's just that the current lenstronomy version in requirements.txt might not reflect it. You can update the lenstronomy requirement to the latest version

**Comment by nataliehogg** [comment]
hi @sibirrer thanks for the comments and sorry that I left this big PR hanging for a week! I didn't have time to get back to it til now. I'll try to finish it today, and thanks for the pointer about the requirements.txt! It seems the tests are passing now and it's just coverage that's missing.

**Comment by nataliehogg** [comment]
hi again @sibirrer I have increased the coverage on this PR, let me know if you want me to try and get it to 100%!

File: jaxtronomy/LensModel/Profiles/sis.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        return f_x, f_y
+
+    def hessian(self, x, y, theta_E, center_x=0, center_y=0):
+        """Returns Hessian matrix of function d^2f/dx^2, d^2/dxdy, d^2/dydx,
+        d^f/dy^2."""
+        x_shift = x - center_x
+        y_shift = y - center_y
+        R = (x_shift * x_shift + y_shift * y_shift) ** (3.0 / 2)
+        if isinstance(R, int) or isinstance(R, float):
+            prefac = theta_E / max(0.000001, R)
```
Comment (sibirrer) [inline jaxtronomy/LensModel/Profiles/sis.py:56]: would be good having this tested to make sure it is JAX compatible (or else find another way to deal with R=0

**Review by sibirrer — CHANGES_REQUESTED** [review]
Hi @nataliehogg thank you very much! I am just unsure how tolerant JAX is for certain code, so having pretty much everything being run through is desired during testing. I have particularly two lines in mind in the SIS profile.

File: jaxtronomy/LensModel/Profiles/sis.py (hunk 1 above)
Comment (nataliehogg) [reply jaxtronomy/LensModel/Profiles/sis.py]: Thanks for the tip @sibirrer, I removed the `isinstance` in favour of `jnp.where` . The tests now cover 100% of the profile.

**Review by sibirrer — APPROVED** [review]
Looks great to me, thank you very much @nataliehogg !

**Pull request merged**